helix-core/src/indent.rs

```diff
@@ -452,11 +490,37 @@ fn query_indents(
                     }
                 }
             }
+            added_indent_captures.push((capture.node.id(), indent_capture))
+        }
+        for (node_id, mut capture) in added_indent_captures {
+            // Set the anchor for all align queries.
+            if let IndentCaptureType::Align(_) = capture.capture_type {
+                let anchor = match anchor {
+                    None => {
+                        log::error!(
+                            "Invalid indent query: @align requires an accompanying @anchor."
+                        );
+                        continue;
+                    }
+                    Some(anchor) => anchor,
+                };
+                // Create a string of tabs & spaces that should have the same width
+                // as the string that precedes the anchor (independent of the tab width).
+                let align = text
+                    .line(anchor.start_position().row)
+                    .byte_slice(0..anchor.start_position().column)
+                    .chars()
+                    .map(|c| match c {
+                        '\t' => '\t',
+                        _ => ' ',
+                    })
```

Comment:
I think this doesn't handle non-ascii and zero width char correctly but I think the right way to handle this would be to

* iterate graphemes instead of chars
* use the graheme width of each char and repeat space that often (except for tabs those stay as is)

its not a huge issue since those are pretty rare but it would be nice to handle correctly regardless

Reply:
Thanks for catching that, I usually don't work a lot with Unicode so I tend to forget these subtleties. Since we already have a grapheme iterator for ropes, this was also a very simple change.